spec/controllers/playlists_controller_spec.rb

```diff
@@ -146,6 +146,53 @@
     end
   end
 
+  describe "update" do
+    let(:user) { users(:jamie_kiesl) }
+
+    before do
+      login(user)
+    end
+
+    it "should update published_at" do
+      expect(playlists(:jamie_kiesl_loves).published_at).to be_nil
+      expect(playlists(:jamie_kiesl_loves).private).to be(true)
+
+      put :update, params: {
+        id: playlists(:jamie_kiesl_loves).id,
+        permalink: 'jamie-loves',
+        user_id: user.login,
+        playlist: {
+          private: false
+        }}
+      expect(playlists(:jamie_kiesl_loves).reload.published_at).not_to be_nil
+    end
+
+    it "should set is_mix to true if is_favorite?" do
+      put :update, params: {
+        id: playlists(:jamie_kiesl_loves).id,
+        permalink: 'jamie-loves',
+        user_id: user.login,
+        playlist: {
+          is_favorite: true
```

Comment:
Bit confused about this — we shouldn't actually be able to do this, so you might have found a bug. There are only 1 favorites playlist per user, it's sort of a "magic" case of playlists.

Reply:
Speaking of "magic" — the playlist creation for favorites happens in a very obscure location: https://github.com/sudara/alonetone/blob/master/app/models/track.rb#L40

We could create a followup where we make that more explicit and maybe even have a favorites_controller that behaves a bit more RESTful. That code is so old...

Reply:
OK. Now I'm confused :)
https://github.com/sudara/alonetone/pull/878/files#diff-8b4a98db7ad82717ddadf5da294a2c1aR139
What does this do then ^?

Reply:
Great Q! In the UI we used to have a stronger definition of mixes (tracks from more than one artists) and albums (tracks from one artist only). 

Favorites, being inherently a mix, were marked as mixes. 

These days we decided not to implement mixes yet. But we may add them soon if artists ask for it again. So although this code was old and broken, it’s nice to keep up the distinction on the backend. 

Since it hasn’t been functional, maybe we need to backfill mixes too, just so the data is clean.

Reply:
This still needs fixing — `is_favorite: true` isn't something that we can pass in the playlist params...